src/dev/build/tasks/fetch_agent_versions_list.ts

```diff
@@ -41,7 +43,9 @@ const getAvailableVersions = async (log: ToolingLog) => {
   } catch (error) {
     const errorMessage = 'Failed to fetch Elastic Agent versions list';
 
-    if (process.env.BUILDKITE_PULL_REQUEST === 'true') {
+    const isPr =
+      !!process.env.BUILDKITE_PULL_REQUEST && process.env.BUILDKITE_PULL_REQUEST !== 'false';
+    if (isPr) {
```

Comment:
We'll also need this check down below on L68 where we skip the task if we're in a PR context

Reply:
updated